apps/system/lockscreen/js/charging.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+(function(exports) {
+
+/**
+ * Creates an object used for refreshing the charging string UI element.
+ * @class  ChargingStatus
+ */
+function ChargingStatus() {
+  this.refreshCB = null;
+}
+
+ChargingStatus.prototype.start = function cs_start() {
+  this.refresh();
+
+  if (!this.refreshCB) {
+    this.refreshCB = this.refresh.bind(this);
+
+    navigator.battery.addEventListener('chargingchange',
+      this.refreshCB);
+    navigator.battery.addEventListener('levelchange',
+      this.refreshCB);
+    navigator.battery.addEventListener('chargingtimechange',
+      this.refreshCB);
+  }
+};
+
+ChargingStatus.prototype.stop = function cs_stop() {
+  if (this.refreshCB) {
```

Comment:
And I don't know why you need to check the `refreshCB` here. If the instance would be started every time we want to use it, there is no chance to have an instance without such method. It's very weird to call the `stop` method of an instance not started, too.